lib/ios.dart

```diff
@@ -1,161 +1,34 @@
 import 'dart:convert';
 import 'dart:io';
+
 import 'package:flutter_launcher_icons/utils.dart';
-import 'package:image/image.dart';
+import 'package:flutter_launcher_icons/icon_template.dart';
 import 'package:flutter_launcher_icons/constants.dart';
+import 'package:flutter_launcher_icons/abstract_platform.dart';
 
-/// File to handle the creation of icons for iOS platform
-class IosIconTemplate {
-  IosIconTemplate({this.size, this.name});
+import 'package:image/image.dart';
 
-  final String name;
-  final int size;
-}
+final IconTemplateGenerator templateGenerator = IconTemplateGenerator(
+    defaultLocation: iosDefaultIconFolder, defaultSuffix: '.png');
 
-List<IosIconTemplate> iosIcons = <IosIconTemplate>[
-  IosIconTemplate(name: '-20x20@1x', size: 20),
-  IosIconTemplate(name: '-20x20@2x', size: 40),
-  IosIconTemplate(name: '-20x20@3x', size: 60),
-  IosIconTemplate(name: '-29x29@1x', size: 29),
-  IosIconTemplate(name: '-29x29@2x', size: 58),
-  IosIconTemplate(name: '-29x29@3x', size: 87),
-  IosIconTemplate(name: '-40x40@1x', size: 40),
-  IosIconTemplate(name: '-40x40@2x', size: 80),
-  IosIconTemplate(name: '-40x40@3x', size: 120),
-  IosIconTemplate(name: '-60x60@2x', size: 120),
-  IosIconTemplate(name: '-60x60@3x', size: 180),
-  IosIconTemplate(name: '-76x76@1x', size: 76),
-  IosIconTemplate(name: '-76x76@2x', size: 152),
-  IosIconTemplate(name: '-83.5x83.5@2x', size: 167),
-  IosIconTemplate(name: '-1024x1024@1x', size: 1024),
+List<IconTemplate> iosIcons = <IconTemplate>[
+  templateGenerator.get(name: '-20x20@1x', size: 20),
+  templateGenerator.get(name: '-20x20@2x', size: 40),
+  templateGenerator.get(name: '-20x20@3x', size: 60),
+  templateGenerator.get(name: '-29x29@1x', size: 29),
+  templateGenerator.get(name: '-29x29@2x', size: 58),
+  templateGenerator.get(name: '-29x29@3x', size: 87),
+  templateGenerator.get(name: '-40x40@1x', size: 40),
+  templateGenerator.get(name: '-40x40@2x', size: 80),
+  templateGenerator.get(name: '-40x40@3x', size: 120),
+  templateGenerator.get(name: '-60x60@2x', size: 120),
+  templateGenerator.get(name: '-60x60@3x', size: 180),
+  templateGenerator.get(name: '-76x76@1x', size: 76),
+  templateGenerator.get(name: '-76x76@2x', size: 152),
+  templateGenerator.get(name: '-83.5x83.5@2x', size: 167),
+  templateGenerator.get(name: '-1024x1024@1x', size: 1024),
 ];
 
-void createIcons(Map<String, dynamic> config, String flavor) {
```

Comment:
Just like in `android.dart`, `createIcons` overrides a method in `AbstractPlatform`. The following code has been moved.